src/Week2/LogAnalysis.hs

```diff
@@ -0,0 +1,104 @@
+{-# OPTIONS_GHC -Wall #-}
+module Week2.LogAnalysis where
+
+import Week2.Log
+
+{-
+Exercise 1
+
+Define a function
+parseMessage :: String -> LogMessage
+which parses an individual line from the log file. 
+
+For example,
+ parseMessage "E 2 562 help help"
+     == LogMessage (Error 2) 562 "help help"
+parseMessage "I 29 la la la"
+     == LogMessage Info 29 "la la la"
+ parseMessage "This is not in the right format"
+     == Unknown "This is not in the right format"
+
+Once we can parse one log message, we can parse a whole log file. 
+Define a function parse :: String -> [LogMessage]
+which parses an entire log file at once and returns its contents as a list of LogMessages.
+-}
+
+parseMessage :: String -> LogMessage
+parseMessage s =
+    let words_list = words s in
+    case words_list of
+    ("I":num:message) -> LogMessage Info (read num) (unwords message)
+    ("W":num:message) -> LogMessage Warning (read num) (unwords message)
+    ("E":val:num:message) -> LogMessage (Error (read val)) (read num) (unwords message)
+    _ -> Unknown (unwords words_list)
+
+parse :: String -> [LogMessage]
+parse s = map parseMessage $ lines s
+
+{-
+Exercise 2 
+
+Define a function
+insert :: LogMessage -> MessageTree -> MessageTree
+which inserts a new LogMessage into an existing MessageTree, pro- ducing a new MessageTree. 
+insert may assume that it is given a sorted MessageTree, and must produce a new sorted MessageTree containing the new LogMessage in addition to the contents of the original MessageTree.
+However, note that if insert is given a LogMessage which is Unknown, it should return the MessageTree unchanged.
+-}
+
+insert :: LogMessage -> MessageTree -> MessageTree
+insert logmsg Leaf = Node Leaf logmsg Leaf
+insert logmsg@(LogMessage _ tmstmp _) (Node left logmsg'@(LogMessage _ tmstmp' _) right)
+    | tmstmp > tmstmp' = Node left logmsg' $ insert logmsg right
+    | otherwise = Node (insert logmsg left) logmsg' right
+insert _ tree = tree
```

Comment:
Style nitpicks: 
- Use camel case for variable names, and make them a bit more descriptive. So `timeStamp` as opposed to `tmstmp`. 
- Avoid using `'` for variable names. Something like `timeStampLeft` would be more readable than `tmstmp'`.